components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -380,10 +381,41 @@ inline void AtmosphereInput::set_degrees_of_freedom()
     m_dofs.emplace(std::make_pair("avg_count",1));
   }
   /* TODO: 
-   * Adjust DOF to accomodate packing for fields 
    * Gather DOF info directly from grid manager
   */
 } // set_degrees_of_freedom
 /* ---------------------------------------------------------- */
+inline std::vector<Int> AtmosphereInput::get_var_dof(const int dof_len, const bool has_cols)
```

Comment:
Do all of these methods _need_ to be inlined? Usually, you inline _very small_ methods (a couple of lines), which are expected to be called quite a few times. If they are long methods and/or not critical for performance, we can simply put their impl in a cpp file. Helps to keep the headers smaller (i.e., easier/faster to parse).